patches/server/0933-Fix-map-cursors.patch

```diff
@@ -0,0 +1,36 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: TheTuso <piotrekpasztor@gmail.com>
+Date: Wed, 10 Aug 2022 19:59:21 +0200
+Subject: [PATCH] Fix map cursors
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/map/CraftMapRenderer.java b/src/main/java/org/bukkit/craftbukkit/map/CraftMapRenderer.java
+index b47f18b1e448807a17ca9f2ae9609680233da837..5ee6a376558a08fc112cfe6e8e93e96b1802dab5 100644
+--- a/src/main/java/org/bukkit/craftbukkit/map/CraftMapRenderer.java
++++ b/src/main/java/org/bukkit/craftbukkit/map/CraftMapRenderer.java
+@@ -1,14 +1,11 @@
+ package org.bukkit.craftbukkit.map;
+ 
++import io.papermc.paper.adventure.PaperAdventure;
+ import net.minecraft.world.level.saveddata.maps.MapDecoration;
+ import net.minecraft.world.level.saveddata.maps.MapItemSavedData;
+ import org.bukkit.Bukkit;
+-import org.bukkit.craftbukkit.util.CraftChatMessage;
+ import org.bukkit.entity.Player;
+-import org.bukkit.map.MapCanvas;
+-import org.bukkit.map.MapCursorCollection;
+-import org.bukkit.map.MapRenderer;
+-import org.bukkit.map.MapView;
++import org.bukkit.map.*;
+ 
+ public class CraftMapRenderer extends MapRenderer {
+ 
+@@ -42,7 +39,7 @@ public class CraftMapRenderer extends MapRenderer {
+             }
+ 
+             MapDecoration decoration = worldMap.decorations.get(key);
+-            cursors.addCursor(decoration.getX(), decoration.getY(), (byte) (decoration.getRot() & 15), decoration.getType().getIcon(), true, CraftChatMessage.fromComponent(decoration.getName()));
++            cursors.addCursor(new MapCursor(decoration.getX(), decoration.getY(), (byte) (decoration.getRot() & 15), decoration.getType().getIcon(), true, decoration.getName() == null ? null : PaperAdventure.asAdventure(decoration.getName()))); // Paper
```

Comment:
Additionally, not much need to create the new MapCursor here.
The constructor is deprecated for the same reason the plain `addCursor` method is deprecated, it uses the magic value from `decoration.getRot()`. This will not change unless you actually parse the api representation of the rotation, which is overkill.

I'd argue to just revert back/move to the `addCursor(int x, int y, byte direction, byte type, boolean visible, @Nullable net.kyori.adventure.text.Component caption)` call

Reply:
✅